### ISSUE TITLE
NextJS hmr not working

### ISSUE THREAD

**Issue by nxy7** [unocss/unocss#2103]
Hello, I'm trying out UnoCSS (great project) and it seems like nextjs HMR is not working even with cache set up as in example

https://github.com/unocss/unocss/blob/bb321caea95dd0a6e0cc44c22d897a9dc96cb6c9/examples/next/next.config.js#L6-L12

I think what's happening is that CSS file is updated, but doesn't get loaded - i see loading spinner but need to refresh the page to see UnoCSS styles updated.

**Comment by eramdam** [comment]
I'm seeing the same behavior with a NextJS 13.x project. I ended up using the CLI similar to what [this article](https://thundermiracle.com/blog/en/2022-06-07-next-with-unocss/) describes 😅

**Comment by RealTong** [comment]
I have the same mistake. (NextJS hmr not working )
```json
    "next": "13.3.0",
    "@unocss/preset-attributify": "^0.50.8",
    "@unocss/preset-uno": "^0.50.8",
    "@unocss/preset-web-fonts": "^0.50.8",
    "@unocss/reset": "^0.50.8",
    "@unocss/webpack": "^0.50.8",
    "unocss": "^0.50.8"
```
```javascript
// next.config.js
const UnoCSS = require('@unocss/webpack').default;

/** @type {import('next').NextConfig} */
const nextConfig = {
    reactStrictMode: true,
    webpack: (config) => {
        config.cache = false
        config.plugins.push(UnoCSS({ presets: [] }));
        return config
    }
}

module.exports = nextConfig

```

```javascript
// unocss.config.js
import {
    defineConfig,
    presetAttributify,
    presetUno,
} from 'unocss';

export default defineConfig({
    presets: [
        presetUno(),
        presetAttributify(),
    ]
})
```

**Comment by sibbng** [comment]
FYI we also have a PostCSS plugin: https://unocss.dev/integrations/postcss

If you don't use any [transformer](https://unocss.dev/presets/#transformers) or only use [transformer-directives](https://unocss.dev/transformers/directives) I would suggest you give it a try. Features of [transformer-directives](https://unocss.dev/transformers/directives) already included in PostCSS plugin, so you don't need to have it in config.

**Comment by RealTong** [comment]
> FYI we also have a PostCSS plugin: https://unocss.dev/integrations/postcss
> 
> If you don't use any [transformer](https://unocss.dev/presets/#transformers) or only use [transformer-directives](https://unocss.dev/transformers/directives) I would suggest you give it a try. Features of [transformer-directives](https://unocss.dev/transformers/directives) already included in PostCSS plugin, so you don't need to have it in config.

I tried postcss according to the documentation.

But I found that it only works at startup. HMR gets stuck with "wait - compiling...".
And in style.css, WebStorm says "Unknown at-rule ".
```javascript
// postcss.config.js
module.exports = {
    plugins: {
        '@unocss/postcss': {
            // Optional
            content: ['**/*.{html,js,ts,jsx,tsx,vue,svelte,astro}'],
        },
    },
}
```

**Comment by sibbng** [comment]
> But I found that it only works at startup. HMR gets stuck with "wait - compiling...".

Which version you are using? I remember that I fixed this issue. If you are using the latest version, could you please create a new issue with a minimal reproduction 🙏

**Comment by RealTong** [comment]
> > But I found that it only works at startup. HMR gets stuck with "wait - compiling...".
> 
> Which version you are using? I remember that I fixed this issue. If you are using the latest version, could you please create a new issue with a minimal reproduction 🙏

I created a minimal demo and new issue in https://github.com/unocss/unocss/issues/2473 .

**Comment by iamhectorsosa** [comment]
>_FYI we also have a PostCSS plugin: https://unocss.dev/integrations/postcss If you don't use any [transformer](https://unocss.dev/presets/#transformers) or only use [transformer-directives](https://unocss.dev/transformers/directives) I would suggest you give it a try. Features of [transformer-directives](https://unocss.dev/transformers/directives) already included in PostCSS plugin, so you don't need to have it in config._

@sibbng what's the alternative then? If we are using PostCSS and need to use transformers? For instance `@unocss/transformer-attributify-jsx`.

**Comment by jacob-8** [comment]
> what's the alternative then? If we are using PostCSS and need to use transformers?

@iamhectorsosa I do want to mention that I think this issue can be closed by your #2724. If there's not a quick alternative, I think the additional use case of attributify should be a separate issue.
